File: data_php/requester.php
<?php

namespace Dropox\Sign\Tester;

use Exception;
use GuzzleHttp;
use Dropbox\Sign as DropboxSign;
use SplFileObject;

ini_set('display_errors', 1);
ini_set('display_startup_errors', 1);
error_reporting(E_ALL);

require_once __DIR__ . '/vendor/autoload.php';

class Requester
{
    private const FILE_UPLOADS_DIR = __DIR__ . '/../file_uploads';

    private string $apiServer;

    private string $authType;

    private string $authKey;

    private array $data;

    private bool $devMode;

    private array $files;

    private string $operationId;

    private array $parameters;

    public function __construct(
        string $authType,
        string $authKey,
        string $apiServer,
        string $jsonData,
        $devMode = null
    ) {
        $this->authType = strtolower($authType);
        $this->authKey = $authKey;
        $this->apiServer = $apiServer;
        $this->devMode = filter_var($devMode, FILTER_VALIDATE_BOOLEAN);

        $this->readJsonData($jsonData);
    }

    public function run()
    {
        try {
            $response = $this->callFromOperationId();

            echo json_encode([
                'body'        => $response[0],
                'status_code' => $response[1],
                'headers'     => $this->getResponseHeaders(
                    $response[2]
                ),
            ], JSON_PRETTY_PRINT);
        } catch (DropboxSign\ApiException $e) {
            echo json_encode([
                'body'        => json_decode($e->getResponseBody()),
                'status_code' => json_decode($e->getCode()),
                'headers'     => $this->getResponseHeaders(
                    $e->getResponseHeaders()
                ),
            ], JSON_PRETTY_PRINT);
        }
    }

    private function getConfig(): DropboxSign\Configuration
    {
        $config = DropboxSign\Configuration::getDefaultConfiguration();
        $config->setHost("https://{$this->apiServer}/v3");

        if ($this->authType === 'apikey') {
            $config->setUsername($this->authKey);
        } elseif ($this->authType === 'oauth') {
            $config->setAccessToken($this->authKey);
        } else {
            throw new Exception(
                'Invalid auth type. Must be "apikey" or "oauth".'
            );
        }

        if (!empty($this->devMode)) {
            $cookie = GuzzleHttp\Cookie\CookieJar::fromArray([
                'XDEBUG_SESSION' => 'xdebug',
            ], $this->apiServer);
            $config->setOptions(['cookies' => $cookie]);
        }

        return $config;
    }

    private function readJsonData(string $base64Json): void
    {
        if (!empty($base64Json)) {
            $json = json_decode(base64_decode($base64Json, true), true);

            if (json_last_error() !== JSON_ERROR_NONE || empty($json)) {
                throw new Exception('Invalid base64 JSON data provided.');
            }
        }

        $this->operationId = $json['operationId'] ?? '';
        $this->data = $json['data'] ?? [];
        $this->files = $json['files'] ?? [];
        $this->parameters = $json['parameters'] ?? [];
    }

    private function callFromOperationId(): array
    {
        if ($response = $this->accountApi()) {
            return $response;
        }
        if ($response = $this->apiAppApi()) {
            return $response;
        }
        if ($response = $this->bulkSendJobApi()) {
            return $response;
        }
        if ($response = $this->embeddedApi()) {
            return $response;
        }
        if ($response = $this->oauthApi()) {
            return $response;
        }
        if ($response = $this->reportApi()) {
            return $response;
        }
        if ($response = $this->signatureRequestApi()) {
            return $response;
        }
        if ($response = $this->teamApi()) {
            return $response;
        }
        if ($response = $this->templateApi()) {
            return $response;
        }
        if ($response = $this->unclaimedDraftApi()) {
            return $response;
        }

        throw new Exception("Invalid operationId: {$this->operationId}");
    }

    private function getResponseHeaders(array $headers): array
    {
        $formatted = [];

        foreach ($headers as $k => $v) {
            if (count($v) > 1) {
                $formatted[strtolower($k)] = $v;

                continue;
            }

            $formatted[strtolower($k)] = array_pop($v);
        }

        return $formatted;
    }

    private function getFile(string $name): ?SplFileObject
    {
        if (!empty($this->files[$name])) {
            return new SplFileObject(
                self::FILE_UPLOADS_DIR . "/{$this->files[$name]}",
            );
        }

        return null;
    }

    /**
     * @return SplFileObject[]
     */
    private function getFiles(string $name): array
    {
        $files = [];

        foreach ($this->files[$name] ?? [] as $file) {
            $files[] = new SplFileObject(
                self::FILE_UPLOADS_DIR . "/{$file}",
            );
        }

        return $files;
    }

    private function accountApi(): ?array
    {
        $api = new DropboxSign\Api\AccountApi(
            $this->getConfig(),
            new GuzzleHttp\Client(),
        );

        if ($this->operationId === 'accountCreate') {
            $obj = DropboxSign\Model\AccountCreateRequest::init(
                $this->data,
            );

            return $api->accountCreateWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'accountGet') {
            return $api->accountGetWithHttpInfo(
                $this->parameters['account_id'] ?? null,
                $this->parameters['email_address'] ?? null,
            );
        }

        if ($this->operationId === 'accountUpdate') {
            $obj = DropboxSign\Model\AccountUpdateRequest::init(
                $this->data,
            );

            return $api->accountUpdateWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'accountVerify') {
            $obj = DropboxSign\Model\AccountVerifyRequest::init(
                $this->data,
            );

            return $api->accountVerifyWithHttpInfo(
                $obj,
            );
        }

        return null;
    }

    private function apiAppApi(): ?array
    {
        $api = new DropboxSign\Api\ApiAppApi(
            $this->getConfig(),
            new GuzzleHttp\Client(),
        );

        if ($this->operationId === 'apiAppCreate') {
            $obj = DropboxSign\Model\ApiAppCreateRequest::init(
                $this->data,
            );
            $obj->setCustomLogoFile($this->getFile('custom_logo_file'));

            return $api->apiAppCreateWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'apiAppDelete') {
            return $api->apiAppDeleteWithHttpInfo(
                $this->parameters['client_id'],
            );
        }

        if ($this->operationId === 'apiAppGet') {
            return $api->apiAppGetWithHttpInfo(
                $this->parameters['client_id'],
            );
        }

        if ($this->operationId === 'apiAppList') {
            return $api->apiAppListWithHttpInfo(
                $this->parameters['page'] ?? 1,
                $this->parameters['page_size'] ?? 20,
            );
        }

        if ($this->operationId === 'apiAppUpdate') {
            $obj = DropboxSign\Model\ApiAppUpdateRequest::init(
                $this->data,
            );
            $obj->setCustomLogoFile($this->getFile('custom_logo_file'));

            return $api->apiAppUpdateWithHttpInfo(
                $this->parameters['client_id'],
                $obj,
            );
        }

        return null;
    }

    private function bulkSendJobApi(): ?array
    {
        $api = new DropboxSign\Api\BulkSendJobApi(
            $this->getConfig(),
            new GuzzleHttp\Client(),
        );

        if ($this->operationId === 'bulkSendJobGet') {
            return $api->bulkSendJobGetWithHttpInfo(
                $this->parameters['bulk_send_job_id'],
            );
        }

        if ($this->operationId === 'bulkSendJobList') {
            return $api->bulkSendJobListWithHttpInfo(
                $this->parameters['page'] ?? 1,
                $this->parameters['page_size'] ?? 20,
            );
        }

        return null;
    }

    private function embeddedApi(): ?array
    {
        $api = new DropboxSign\Api\EmbeddedApi(
            $this->getConfig(),
            new GuzzleHttp\Client(),
        );

        if ($this->operationId === 'embeddedEditUrl') {
            $obj = DropboxSign\Model\EmbeddedEditUrlRequest::init(
                $this->data,
            );

            return $api->embeddedEditUrlWithHttpInfo(
                $this->parameters['template_id'],
                $obj,
            );
        }

        if ($this->operationId === 'embeddedSignUrl') {
            return $api->embeddedSignUrlWithHttpInfo(
                $this->parameters['signature_id'],
            );
        }

        return null;
    }

    private function oauthApi(): ?array
    {
        $api = new DropboxSign\Api\OAuthApi(
            $this->getConfig(),
            new GuzzleHttp\Client(),
        );

        if ($this->operationId === 'oauthTokenGenerate') {
            $obj = DropboxSign\Model\OAuthTokenGenerateRequest::init(
                $this->data,
            );

            return $api->oauthTokenGenerateWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'oauthTokenRefresh') {
            $obj = DropboxSign\Model\OAuthTokenRefreshRequest::init(
                $this->data,
            );

            return $api->oauthTokenRefreshWithHttpInfo(
                $obj,
            );
        }

        return null;
    }

    private function reportApi(): ?array
    {
        $api = new DropboxSign\Api\ReportApi(
            $this->getConfig(),
            new GuzzleHttp\Client(),
        );

        if ($this->operationId === 'reportCreate') {
            $obj = DropboxSign\Model\ReportCreateRequest::init(
                $this->data,
            );

            return $api->reportCreateWithHttpInfo(
                $obj,
            );
        }

        return null;
    }

    private function signatureRequestApi(): ?array
    {
        $api = new DropboxSign\Api\SignatureRequestApi(
            $this->getConfig(),
            new GuzzleHttp\Client(),
        );

        if ($this->operationId === 'signatureRequestBulkCreateEmbeddedWithTemplate') {
            $obj = DropboxSign\Model\SignatureRequestBulkCreateEmbeddedWithTemplateRequest::init(
                $this->data,
            );

            $obj->setSignerFile($this->getFile('signer_file'));

            return $api->signatureRequestBulkCreateEmbeddedWithTemplateWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'signatureRequestBulkSendWithTemplate') {
            $obj = DropboxSign\Model\SignatureRequestBulkSendWithTemplateRequest::init(
                $this->data,
            );

            $obj->setSignerFile($this->getFile('signer_file'));

            return $api->signatureRequestBulkSendWithTemplateWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'signatureRequestCancel') {
            return $api->signatureRequestCancelWithHttpInfo(
                $this->parameters['signature_request_id'],
            );
        }

        if ($this->operationId === 'signatureRequestCreateEmbedded') {
            $obj = DropboxSign\Model\SignatureRequestCreateEmbeddedRequest::init(
                $this->data,
            );
            $obj->setFiles($this->getFiles('files'));

            return $api->signatureRequestCreateEmbeddedWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'signatureRequestCreateEmbeddedWithTemplate') {
            $obj = DropboxSign\Model\SignatureRequestCreateEmbeddedWithTemplateRequest::init(
                $this->data,
            );
            $obj->setFiles($this->getFiles('files'));

            return $api->signatureRequestCreateEmbeddedWithTemplateWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'signatureRequestFilesAsFileUrl') {
            return $api->signatureRequestFilesAsFileUrlWithHttpInfo(
                $this->parameters['signature_request_id']
            );
        }

        if ($this->operationId === 'signatureRequestGet') {
            return $api->signatureRequestGetWithHttpInfo(
                $this->parameters['signature_request_id'],
            );
        }

        if ($this->operationId === 'signatureRequestList') {
            return $api->signatureRequestListWithHttpInfo(
                $this->parameters['account_id'] ?? null,
                $this->parameters['page'] ?? 1,
                $this->parameters['page_size'] ?? 20,
                $this->parameters['query'] ?? null,
            );
        }

        if ($this->operationId === 'signatureRequestReleaseHold') {
            return $api->signatureRequestReleaseHoldWithHttpInfo(
                $this->parameters['signature_request_id'],
            );
        }

        if ($this->operationId === 'signatureRequestRemind') {
            $obj = DropboxSign\Model\SignatureRequestRemindRequest::init(
                $this->data,
            );

            return $api->signatureRequestRemindWithHttpInfo(
                $this->parameters['signature_request_id'],
                $obj,
            );
        }

        if ($this->operationId === 'signatureRequestRemove') {
            return $api->signatureRequestRemoveWithHttpInfo(
                $this->parameters['signature_request_id'],
            );
        }

        if ($this->operationId === 'signatureRequestSend') {
            $obj = DropboxSign\Model\SignatureRequestSendRequest::init(
                $this->data,
            );
            $obj->setFiles($this->getFiles('files'));

            return $api->signatureRequestSendWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'signatureRequestSendWithTemplate') {
            $obj = DropboxSign\Model\SignatureRequestSendWithTemplateRequest::init(
                $this->data,
            );
            $obj->setFiles($this->getFiles('files'));

            return $api->signatureRequestSendWithTemplateWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'signatureRequestUpdate') {
            $obj = DropboxSign\Model\SignatureRequestUpdateRequest::init(
                $this->data,
            );

            return $api->signatureRequestUpdateWithHttpInfo(
                $this->parameters['signature_request_id'],
                $obj,
            );
        }

        return null;
    }

    private function teamApi(): ?array
    {
        $api = new DropboxSign\Api\TeamApi(
            $this->getConfig(),
            new GuzzleHttp\Client(),
        );

        if ($this->operationId === 'teamAddMember') {
            $obj = DropboxSign\Model\TeamAddMemberRequest::init(
                $this->data,
            );

            return $api->teamAddMemberWithHttpInfo(
                $obj,
                $this->parameters['team_id'] ?? null,
            );
        }

        if ($this->operationId === 'teamCreate') {
            $obj = DropboxSign\Model\TeamCreateRequest::init(
                $this->data,
            );

            return $api->teamCreateWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'teamDelete') {
            return $api->teamDeleteWithHttpInfo();
        }

        if ($this->operationId === 'teamGet') {
            return $api->teamGetWithHttpInfo();
        }

        if ($this->operationId === 'teamRemoveMember') {
            $obj = DropboxSign\Model\TeamRemoveMemberRequest::init(
                $this->data,
            );

            return $api->teamRemoveMemberWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'teamUpdate') {
            $obj = DropboxSign\Model\TeamUpdateRequest::init(
                $this->data,
            );

            return $api->teamUpdateWithHttpInfo(
                $obj,
            );
        }

        return null;
    }

    private function templateApi(): ?array
    {
        $api = new DropboxSign\Api\TemplateApi(
            $this->getConfig(),
            new GuzzleHttp\Client(),
        );

        if ($this->operationId === 'templateAddUser') {
            $obj = DropboxSign\Model\TemplateAddUserRequest::init(
                $this->data,
            );

            return $api->templateAddUserWithHttpInfo(
                $this->parameters['template_id'],
                $obj,
            );
        }

        if ($this->operationId === 'templateCreateEmbeddedDraft') {
            $obj = DropboxSign\Model\TemplateCreateEmbeddedDraftRequest::init(
                $this->data,
            );
            $obj->setFiles($this->getFiles('files'));

            return $api->templateCreateEmbeddedDraftWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'templateDelete') {
            return $api->templateDeleteWithHttpInfo(
                $this->parameters['template_id'],
            );
        }

        if ($this->operationId === 'templateFilesAsFileUrl') {
            return $api->templateFilesAsFileUrlWithHttpInfo(
                $this->parameters['template_id']
            );
        }

        if ($this->operationId === 'templateGet') {
            return $api->templateGetWithHttpInfo(
                $this->parameters['template_id'],
            );
        }

        if ($this->operationId === 'templateList') {
            return $api->templateListWithHttpInfo(
                $this->parameters['account_id'] ?? null,
                $this->parameters['page'] ?? 1,
                $this->parameters['page_size'] ?? 20,
                $this->parameters['query'] ?? null,
            );
        }

        if ($this->operationId === 'templateRemoveUser') {
            $obj = DropboxSign\Model\TemplateRemoveUserRequest::init(
                $this->data,
            );

            return $api->templateRemoveUserWithHttpInfo(
                $this->parameters['template_id'],
                $obj,
            );
        }

        if ($this->operationId === 'templateUpdateFiles') {
            $obj = DropboxSign\Model\TemplateUpdateFilesRequest::init(
                $this->data,
            );
            $obj->setFiles($this->getFiles('files'));

            return $api->templateUpdateFilesWithHttpInfo(
                $this->parameters['template_id'],
                $obj,
            );
        }

        return null;
    }

    private function unclaimedDraftApi(): ?array
    {
        $api = new DropboxSign\Api\UnclaimedDraftApi(
            $this->getConfig(),
            new GuzzleHttp\Client(),
        );

        if ($this->operationId === 'unclaimedDraftCreate') {
            $obj = DropboxSign\Model\UnclaimedDraftCreateRequest::init(
                $this->data,
            );
            $obj->setFiles($this->getFiles('files'));

            return $api->unclaimedDraftCreateWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'unclaimedDraftCreateEmbedded') {
            $obj = DropboxSign\Model\UnclaimedDraftCreateEmbeddedRequest::init(
                $this->data,
            );
            $obj->setFiles($this->getFiles('files'));

            return $api->unclaimedDraftCreateEmbeddedWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'unclaimedDraftCreateEmbeddedWithTemplate') {
            $obj = DropboxSign\Model\UnclaimedDraftCreateEmbeddedWithTemplateRequest::init(
                $this->data,
            );
            $obj->setFiles($this->getFiles('files'));

            return $api->unclaimedDraftCreateEmbeddedWithTemplateWithHttpInfo(
                $obj,
            );
        }

        if ($this->operationId === 'unclaimedDraftEditAndResend') {
            $obj = DropboxSign\Model\UnclaimedDraftEditAndResendRequest::init(
                $this->data,
            );

            return $api->unclaimedDraftEditAndResendWithHttpInfo(
                $this->parameters['signature_request_id'],
                $obj,
            );
        }

        return null;
    }
}

$requester = new Requester(
    getenv('AUTH_TYPE'),
    getenv('AUTH_KEY'),
    getenv('API_SERVER'),
    getenv('JSON_DATA'),
    getenv('DEV_MODE'),
);

$requester->run();

File: data_python/requester.py
import base64
import json
import os

from dropbox_sign import ApiClient, Configuration, ApiException, apis, models as m


class Requester(object):
    FILE_UPLOADS_DIR = './../file_uploads'

    def __init__(
            self,
            auth_type: str,
            auth_key: str,
            api_server: str,
            json_data: str = None,
            dev_mode: str = None
    ):
        self._auth_type = auth_type
        self._auth_key = auth_key
        self._api_server = api_server
        self._dev_mode = bool(dev_mode)

        self._operation_id = ''
        self._data = {}
        self._files = {}
        self._parameters = {}

        self._read_json_data(json_data)
        self._api_client = ApiClient(self._get_config())

        if self._dev_mode:
            self._api_client.set_default_header('Cookie', 'XDEBUG_SESSION=xdebug')

    def run(self):
        try:
            response = self._call_from_operation_id()

            data = {
                'body': response.data.to_dict() if response.data is not None else None,
                'status_code': response.status_code,
                'headers': self._get_response_headers(response.headers or {}),
            }
        except ApiException as e:
            data = {
                'body': self._exception_body(e),
                'status_code': e.status,
                'headers': self._get_response_headers(e.headers or {}),
            }

        print(json.dumps(data, indent=4))

    @staticmethod
    def _exception_body(exc: ApiException):
        if getattr(exc, 'data', None) is not None and hasattr(exc.data, 'to_dict'):
            return exc.data.to_dict()
        body = getattr(exc, 'body', None)
        if isinstance(body, (bytes, bytearray)):
            try:
                body = body.decode('utf-8')
            except UnicodeDecodeError:
                return None
        if isinstance(body, str) and body:
            try:
                return json.loads(body)
            except json.JSONDecodeError:
                return {'raw': body}
        return None

    def _get_config(self):
        if self._auth_type == 'apikey':
            config = Configuration(
                host=f'https://{self._api_server}/v3',
                username=self._auth_key,
            )
        elif self._auth_type == 'oauth':
            config = Configuration(
                host=f'https://{self._api_server}/v3',
                access_token=self._auth_key,
            )
        else:
            raise RuntimeError('Invalid auth type. Must be "apikey" or "oauth".')

        return config

    def _read_json_data(self, base64_json: str):
        if isinstance(base64_json, str) and len(base64_json):
            json_data = json.loads(base64.b64decode(base64_json))

            if not json_data:
                raise RuntimeError('Invalid base64 JSON data provided.')
        else:
            raise RuntimeError

        if 'operationId' in json_data:
            self._operation_id = json_data['operationId']
        if 'data' in json_data:
            self._data = json_data['data']
        if 'files' in json_data:
            self._files = json_data['files']
        if 'parameters' in json_data:
            self._parameters = json_data['parameters']

    def _call_from_operation_id(self):
        response = self._account_api()
        if response:
            return response

        response = self._api_app_api()
        if response:
            return response

        response = self._bulk_send_job_api()
        if response:
            return response

        response = self._embedded_api()
        if response:
            return response

        response = self._oauth_api()
        if response:
            return response

        response = self._report_api()
        if response:
            return response

        response = self._signature_request_api()
        if response:
            return response

        response = self._team_api()
        if response:
            return response

        response = self._template_api()
        if response:
            return response

        response = self._unclaimed_draft_api()
        if response:
            return response

        raise RuntimeError(f'Invalid operationId: {self._operation_id}')

    @staticmethod
    def _get_response_headers(headers):
        formatted = {}

        if not headers:
            return formatted

        items = headers.items() if hasattr(headers, 'items') else headers
        for key, value in items:
            formatted[key.lower()] = value

        return formatted

    def _get_file(self, name: str):
        if name in self._files and len(self._files[name]):
            f = open(f'{self.FILE_UPLOADS_DIR}/{self._files[name]}', 'rb')
            return f

    def _get_files(self, name: str):
        files = []

        if name in self._files and len(self._files[name]):
            for file in self._files[name]:
                f = open(f'{self.FILE_UPLOADS_DIR}/{file}', 'rb')
                files.append(f)

        return files

    def _account_api(self):
        api = apis.AccountApi(self._api_client)

        if self._operation_id == 'accountCreate':
            obj = m.AccountCreateRequest.init(self._data)

            return api.account_create_with_http_info(obj)

        if self._operation_id == 'accountGet':
            return api.account_get_with_http_info(
                account_id=self._parameters.get('account_id', None),
                email_address=self._parameters.get('email_address', None),
            )

        if self._operation_id == 'accountUpdate':
            obj = m.AccountUpdateRequest.init(self._data)

            return api.account_update_with_http_info(obj)

        if self._operation_id == 'accountVerify':
            obj = m.AccountVerifyRequest.init(self._data)

            return api.account_verify_with_http_info(obj)

    def _api_app_api(self):
        api = apis.ApiAppApi(self._api_client)

        if self._operation_id == 'apiAppCreate':
            obj = m.ApiAppCreateRequest.init(self._data)
            obj.custom_logo_file = self._get_file('custom_logo_file')

            return api.api_app_create_with_http_info(obj)

        if self._operation_id == 'apiAppDelete':
            return api.api_app_delete_with_http_info(
                self._parameters.get('client_id'),
            )

        if self._operation_id == 'apiAppGet':
            return api.api_app_get_with_http_info(
                self._parameters.get('client_id'),
            )

        if self._operation_id == 'apiAppList':
            return api.api_app_list_with_http_info(
                page=self._parameters.get('page', 1),
                page_size=self._parameters.get('page_size', 20),
            )

        if self._operation_id == 'apiAppUpdate':
            obj = m.ApiAppUpdateRequest.init(self._data)
            obj.custom_logo_file = self._get_file('custom_logo_file')

            return api.api_app_update_with_http_info(
                self._parameters.get('client_id'),
                obj,
            )

    def _bulk_send_job_api(self):
        api = apis.BulkSendJobApi(self._api_client)

        if self._operation_id == 'bulkSendJobGet':
            return api.bulk_send_job_get_with_http_info(
                self._parameters.get('bulk_send_job_id'),
            )

        if self._operation_id == 'bulkSendJobList':
            return api.bulk_send_job_list_with_http_info(
                page=self._parameters.get('page', 1),
                page_size=self._parameters.get('page_size', 20),
            )

    def _embedded_api(self):
        api = apis.EmbeddedApi(self._api_client)

        if self._operation_id == 'embeddedEditUrl':
            obj = m.EmbeddedEditUrlRequest.init(self._data)

            return api.embedded_edit_url_with_http_info(
                self._parameters.get('template_id'),
                obj,
            )

        if self._operation_id == 'embeddedSignUrl':
            return api.embedded_sign_url_with_http_info(
                self._parameters.get('signature_id'),
            )

    def _oauth_api(self):
        api = apis.OAuthApi(self._api_client)

        if self._operation_id == 'oauthTokenGenerate':
            obj = m.OAuthTokenGenerateRequest.init(self._data)

            return api.oauth_token_generate_with_http_info(obj)

        if self._operation_id == 'oauthTokenRefresh':
            obj = m.OAuthTokenRefreshRequest.init(self._data)

            return api.oauth_token_refresh_with_http_info(obj)

    def _report_api(self):
        api = apis.ReportApi(self._api_client)

        if self._operation_id == 'reportCreate':
            obj = m.ReportCreateRequest.init(self._data)

            return api.report_create_with_http_info(obj)

    def _signature_request_api(self):
        api = apis.SignatureRequestApi(self._api_client)

        if self._operation_id == 'signatureRequestBulkCreateEmbeddedWithTemplate':
            obj = m.SignatureRequestBulkCreateEmbeddedWithTemplateRequest.init(self._data)
            obj.signer_file = self._get_file('signer_file')

            return api.signature_request_bulk_create_embedded_with_template_with_http_info(obj)

        if self._operation_id == 'signatureRequestBulkSendWithTemplate':
            obj = m.SignatureRequestBulkSendWithTemplateRequest.init(self._data)
            obj.signer_file = self._get_file('signer_file')

            return api.signature_request_bulk_send_with_template_with_http_info(obj)

        if self._operation_id == 'signatureRequestCancel':
            return api.signature_request_cancel_with_http_info(
                self._parameters.get('signature_request_id'),
            )

        if self._operation_id == 'signatureRequestCreateEmbedded':
            obj = m.SignatureRequestCreateEmbeddedRequest.init(self._data)
            obj.files = self._get_files('files')

            return api.signature_request_create_embedded_with_http_info(obj)

        if self._operation_id == 'signatureRequestCreateEmbeddedWithTemplate':
            obj = m.SignatureRequestCreateEmbeddedWithTemplateRequest.init(self._data)
            obj.files = self._get_files('files')

            return api.signature_request_create_embedded_with_template_with_http_info(obj)

        if self._operation_id == 'signatureRequestFilesAsFileUrl':
            return api.signature_request_files_as_file_url_with_http_info(
                self._parameters.get('signature_request_id'),
            )

        if self._operation_id == 'signatureRequestGet':
            return api.signature_request_get_with_http_info(
                self._parameters.get('signature_request_id'),
            )

        if self._operation_id == 'signatureRequestList':
            return api.signature_request_list_with_http_info(
                account_id=self._parameters.get('account_id', None),
                page=self._parameters.get('page', 1),
                page_size=self._parameters.get('page_size', 20),
                query=self._parameters.get('query', None),
            )

        if self._operation_id == 'signatureRequestReleaseHold':
            return api.signature_request_release_hold_with_http_info(
                self._parameters.get('signature_request_id'),
            )

        if self._operation_id == 'signatureRequestRemind':
            obj = m.SignatureRequestRemindRequest.init(self._data)

            return api.signature_request_remind_with_http_info(
                self._parameters.get('signature_request_id'),
                obj,
            )

        if self._operation_id == 'signatureRequestRemove':
            return api.signature_request_remove_with_http_info(
                self._parameters.get('signature_request_id'),
            )

        if self._operation_id == 'signatureRequestSend':
            obj = m.SignatureRequestSendRequest.init(self._data)
            obj.files = self._get_files('files')

            return api.signature_request_send_with_http_info(obj)

        if self._operation_id == 'signatureRequestSendWithTemplate':
            obj = m.SignatureRequestSendWithTemplateRequest.init(self._data)
            obj.files = self._get_files('files')

            return api.signature_request_send_with_template_with_http_info(obj)

        if self._operation_id == 'signatureRequestUpdate':
            obj = m.SignatureRequestUpdateRequest.init(self._data)

            return api.signature_request_update_with_http_info(
                self._parameters.get('signature_request_id'),
                obj,
            )

    def _team_api(self):
        api = apis.TeamApi(self._api_client)

        if self._operation_id == 'teamAddMember':
            obj = m.TeamAddMemberRequest.init(self._data)

            return api.team_add_member_with_http_info(
                obj,
                team_id=self._parameters.get('team_id', None),
            )

        if self._operation_id == 'teamCreate':
            obj = m.TeamCreateRequest.init(self._data)

            return api.team_create_with_http_info(obj)

        if self._operation_id == 'teamDelete':
            return api.team_delete_with_http_info()

        if self._operation_id == 'teamGet':
            return api.team_get_with_http_info()

        if self._operation_id == 'teamRemoveMember':
            obj = m.TeamRemoveMemberRequest.init(self._data)

            return api.team_update_with_http_info(obj)

        if self._operation_id == 'teamUpdate':
            obj = m.TeamUpdateRequest.init(self._data)

            return api.team_update_with_http_info(obj)

    def _template_api(self):
        api = apis.TemplateApi(self._api_client)

        if self._operation_id == 'templateAddUser':
            obj = m.TemplateAddUserRequest.init(self._data)

            return api.template_add_user_with_http_info(
                self._parameters.get('template_id', None),
                obj,
            )

        if self._operation_id == 'templateCreateEmbeddedDraft':
            obj = m.TemplateCreateEmbeddedDraftRequest.init(self._data)
            obj.files = self._get_files('files')

            return api.template_create_embedded_draft_with_http_info(obj)

        if self._operation_id == 'templateDelete':
            return api.template_delete_with_http_info(
                self._parameters.get('template_id'),
            )

        if self._operation_id == 'templateFilesAsFileUrl':
            return api.template_files_as_file_url_with_http_info(
                self._parameters.get('template_id'),
            )

        if self._operation_id == 'templateGet':
            return api.template_get_with_http_info(
                self._parameters.get('template_id'),
            )

        if self._operation_id == 'templateList':
            return api.template_list_with_http_info(
                account_id=self._parameters.get('account_id', None),
                page=self._parameters.get('page', 1),
                page_size=self._parameters.get('page_size', 20),
                query=self._parameters.get('query', None),
            )

        if self._operation_id == 'templateRemoveUser':
            obj = m.TemplateRemoveUserRequest.init(self._data)

            return api.template_remove_user_with_http_info(
                self._parameters.get('template_id'),
                obj,
            )

        if self._operation_id == 'templateUpdateFiles':
            obj = m.TemplateUpdateFilesRequest.init(self._data)
            obj.files = self._get_files('files')

            return api.template_update_files_with_http_info(
                self._parameters.get('template_id'),
                obj,
            )

    def _unclaimed_draft_api(self):
        api = apis.UnclaimedDraftApi(self._api_client)

        if self._operation_id == 'unclaimedDraftCreate':
            obj = m.UnclaimedDraftCreateRequest.init(self._data)
            obj.files = self._get_files('files')

            return api.unclaimed_draft_create_with_http_info(obj)

        if self._operation_id == 'unclaimedDraftCreateEmbedded':
            obj = m.UnclaimedDraftCreateEmbeddedRequest.init(self._data)
            obj.files = self._get_files('files')

            return api.unclaimed_draft_create_embedded_with_http_info(obj)

        if self._operation_id == 'unclaimedDraftCreateEmbeddedWithTemplate':
            obj = m.UnclaimedDraftCreateEmbeddedWithTemplateRequest.init(self._data)
            obj.files = self._get_files('files')

            return api.unclaimed_draft_create_embedded_with_template_with_http_info(obj)

        if self._operation_id == 'unclaimedDraftEditAndResend':
            obj = m.UnclaimedDraftEditAndResendRequest.init(self._data)

            return api.unclaimed_draft_edit_and_resend_with_http_info(
                self._parameters.get('signature_request_id'),
                obj,
            )


if __name__ == '__main__':
    requester = Requester(
        os.getenv('AUTH_TYPE'),
        os.getenv('AUTH_KEY'),
        os.getenv('API_SERVER'),
        os.getenv('JSON_DATA'),
        os.getenv('DEV_MODE'),
    )
    requester.run()
